File: 1291. Sequential Digits/Approach03.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> sequentialDigits(int low, int high) {
        vector<int> result;
        queue<int> q;

        for(int i=1; i<=8; i++){
            q.push(i);
        }

        while(!q.empty()){
            int val = q.front();
            q.pop();
            if(val > high){break;}
            if(val >= low and val <= high){
                result.push_back(val);
            }

            int next_digit = val%10 + 1;
            if(next_digit <= 9){
                q.push(val*10 + next_digit);
            }
        }
        return result;
    }
};

File: 1291. Sequential Digits/Approach03.py
from typing import List
from collections import deque

class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        result = []
        q = deque()

        for i in range(1, 9):
            q.append(i)

        while q:
            val = q.popleft()

            if val > high:
                return result

            if low <= val <= high:
                result.append(val)

            next_val = val % 10 + 1
            if next_val <= 9:
                q.append(val * 10 + next_val)

        return result
